addons/payment_mollie/models/payment_transaction.py

```diff
@@ -129,3 +164,59 @@ def _apply_updates(self, payment_data):
                 payment_status, self.reference
             )
             self._set_error(_("Received data with invalid payment status: %s.", payment_status))
+
+    def _extract_token_values(self, payment_data):
+        """Return dict suitable for creating a payment.token from Mollie payment data."""
+        if self.provider_code != 'mollie':
+            return super()._extract_token_values(payment_data)
+
+        mandate_id = payment_data.get('mandateId')
+        if not mandate_id:
+            return {}
+
+        details = payment_data.get('details', {}) or {}
+        card_number = details.get('cardNumber')
+        payment_details = None
+        if card_number:
+            payment_details = f"{str(card_number)[-4:]}" if len(str(card_number)) >= 4 else card_number
```

Comment:
in the docs they send string already, no?
and we can cut always last 4 chars
```suggestion
            payment_details = str(card_number[-4:])
```